src/Schema/Column/ColumnFactory.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Schema\Column;
+
+use Yiisoft\Db\Schema\SchemaInterface;
+
+use function explode;
+use function preg_match;
+use function str_ireplace;
+use function stripos;
+use function strlen;
+use function strtolower;
+use function substr;
+use function trim;
+
+use const PHP_INT_SIZE;
+
+/**
+ * The default implementation of the {@see ColumnFactoryInterface}.
+ *
+ * @psalm-import-type ColumnInfo from ColumnSchemaInterface
+ * @psalm-suppress MixedArgumentTypeCoercion
+ */
+class ColumnFactory implements ColumnFactoryInterface
```

Comment:
```suggestion
abstract class AbstractColumnFactory implements ColumnFactoryInterface
```

All DBMS inherit column factory. Suggest mark this class abstract.

Reply:
`ColumnBuilder` will be made to generate columns when creating tables, for example in `db-migration` package or in `db` package. Regardless of the specific DBMS. For this, an independent implementation of `ColumnFactory` is needed.

Reply:
Perhaps make sense to realize separate implementation from `AbstractColumnFactory` for this purpose.